src/internal.c

```diff
@@ -8558,6 +8701,7 @@ int ProcessPeerCerts(WOLFSSL* ssl, byte* input, word32* inOutIdx,
 
                 if (args->fatal) {
                     ssl->error = ret;
+                    SendAlert(ssl, alert_fatal, bad_certificate);
```

Comment:
Can you provide some context for why this SendAlert was added? It does seem appropriate to send this, however there might be a reason we do not. However I do see we send the same alert in the always verify case if the callbacks indicates a failure. So this seems okay, but curious why you decided to add this? If we are concerned about backwards compatibility maybe only enable this for OPENSSL_EXTRA or use a new define?

Reply:
I believe that this was added upon request from the client for some expected behavior compatibility when encountering bad certificates. Adding OPENSSL_EXTRA wrappers may be good here.